lib/Dialect/TritonGPU/Transforms/LoopScheduling.cpp

```diff
@@ -23,6 +23,29 @@ namespace gpu {
 
 namespace {
 
+bool hasGpuBarriers(scf::ForOp forOp) {
+  for (auto &op : forOp.getBody()->without_terminator()) {
```

Comment:
Needs to be `forOp.walk` otherwise you would miss barriers inside of control flow regions.

Reply:
Good catch!